boards/x86/up_squared/board.cmake

```diff
@@ -0,0 +1,4 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_set_flasher_ifnset(misc-flasher)
+board_finalize_runner_args(misc-flasher)
```

Comment:
This is very interesting... I guess you're calling sanitycheck with the complete path to your script and all its arguments?

I'm not sure why we need to involve west in this, perhaps sanitycheck can just take an option for a custom flash command.

But to keep west involved, I suggested in another PR that we could have misc-flasher set some environment variables to contain at least the zephyr build directory, if not also the bin / hex / elf files inside it. Will that help you?

Reply:
yes, i call the sanitycheck with my scripts' path and needed arguments. 
becasue sanitycheck has a --west-flash option which can be used to pass my scripts and arguments.

'custom flash'? do you mean the 'make flash'? if i use the make flash, i can't pass my arguments, because the '--west-flash' option is only for the west flash command.

great. if  i know the path of zephyr build directory, i can use my script to search for the test image path in it.
for now, the problem is that I can't determine the test image path when i run sanitycheck, so if i know the build directory, it will be fine.